base/iterators.jl

```diff
@@ -1170,6 +1170,8 @@ flatten_iteratorsize(a, b) = SizeUnknown()
 _flatten_iteratorsize(sz, ::EltypeUnknown, I) = SizeUnknown()
 _flatten_iteratorsize(sz, ::HasEltype, I) = flatten_iteratorsize(sz, eltype(I))
 _flatten_iteratorsize(sz, ::HasEltype, ::Type{Tuple{}}) = HasLength()
+_flatten_iteratorsize(sz, ::HasEltype, ::Type{<:Tuple{Vararg{AbstractArray}}}) = HasLength()
+_flatten_iteratorsize(sz, ::HasEltype, ::Type{<:AbstractArray{<:AbstractArray}}) = HasLength()
```

Comment:
I don't see how `HasEltype` is relevant here, would this work too?
```suggestion
_flatten_iteratorsize(sz, _, ::Type{<:Tuple{Vararg{AbstractArray}}}) = HasLength()
_flatten_iteratorsize(sz, _, ::Type{<:AbstractArray{<:AbstractArray}}) = HasLength()
```